### ISSUE TITLE
test-dhcp-client: Assertion failures

### ISSUE THREAD

**Issue by YHNdnzj** [systemd/systemd#28990]
### systemd version the issue has been seen with

latest main

### Unexpected behaviour you saw

https://github.com/systemd/systemd/actions/runs/5998149754/job/16265878766?pr=28987

https://github.com/systemd/systemd/actions/runs/5998181339/job/16265956748?pr=28988

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
450/1522 systemd:network / test-dhcp-client                                                     FAIL             2.20s   killed by signal 6 SIGABRT
>>> LD_LIBRARY_PATH=/home/runner/work/systemd/systemd/build/src/shared PATH=/home/runner/work/systemd/systemd/build:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin EFI_ADDON=/home/runner/work/systemd/systemd/build/src/boot/efi/addonx64.efi.stub MALLOC_PERTURB_=61 PROJECT_BUILD_ROOT=/home/runner/work/systemd/systemd/build SYSTEMD_LANGUAGE_FALLBACK_MAP=/home/runner/work/systemd/systemd/src/locale/language-fallback-map /home/runner/work/systemd/systemd/build/test-dhcp-client
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stderr:
Assertion 'client' failed at src/libsystemd-network/sd-dhcp-client.c:252, function sd_dhcp_client_set_request_option(). Ignoring.
Assertion 'client' failed at src/libsystemd-network/sd-dhcp-client.c:275, function sd_dhcp_client_set_request_address(). Ignoring.
Assertion 'client' failed at src/libsystemd-network/sd-dhcp-client.c:287, function sd_dhcp_client_set_ifindex(). Ignoring.
Assertion 'ifindex > 0' failed at src/libsystemd-network/sd-dhcp-client.c:289, function sd_dhcp_client_set_ifindex(). Ignoring.
Assertion 'ifindex > 0' failed at src/libsystemd-network/sd-dhcp-client.c:289, function sd_dhcp_client_set_ifindex(). Ignoring.
Assertion 'ifindex > 0' failed at src/libsystemd-network/sd-dhcp-client.c:289, function sd_dhcp_client_set_ifindex(). Ignoring.
lo: DHCPv4 client: FREE
DHCPv4 client: FREE
DHCPv4 client: STARTED on ifindex 42
Assertion '!sd_dhcp_client_is_running(client)' failed at src/libsystemd-network/sd-dhcp-client.c:489, function sd_dhcp_client_set_iaid_duid_en(). Ignoring.
DHCPv4 client: STOPPED
DHCPv4 client: FREE
DHCPv4 client: STARTED on ifindex 42
Assertion '!sd_dhcp_client_is_running(client)' failed at src/libsystemd-network/sd-dhcp-client.c:489, function sd_dhcp_client_set_iaid_duid_en(). Ignoring.
Assertion 'sd_event_loop(e) >= 0' failed at src/libsystemd-network/test-dhcp-client.c:525, function test_addr_acq(). Aborting.
```

**Comment by bluca** [comment]
It's been introduced by https://github.com/systemd/systemd/pull/28932 but it's strange because the CI is green there. Temporary workaround here https://github.com/systemd/systemd/pull/28991 so that it can be investigated without rush

**Comment by yuwata** [comment]
@bluca Thank you for adding workaround quickly. The issue should be fixed by #29013. PTAL.
